### DISCUSSION TITLE
help request: sometimes the headers recorded in ctx are not what I am currently requesting

### DISCUSSION THREAD

**Discussion by amZhan** [apache/apisix#7673] [Q&A]
### Description

I wrote an authentication plug-in myself to obtain some business interface permissions by taking header information, but sometimes the headers recorded in ctx are not what I am currently requesting, and it feels like the header information requested by others last time.

### Environment

- APISIX version (run `apisix version`):2.9
- Operating system (run `uname -a`):centos7.9
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):1.19.9.1
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):3.5
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

**Comment by tzssangglass** [comment]
You should give the code directly and give the verification method.
